krabs/krabs/ut.hpp

```diff
@@ -99,42 +105,79 @@ namespace krabs { namespace details {
     inline void ut::enable_providers(
         const krabs::trace<krabs::details::ut> &trace)
     {
-        std::map<krabs::guid, std::tuple<UCHAR, ULONGLONG, ULONGLONG, UCHAR>> providerFlags;
+        provider_filter_settings provider_flags;
 
         // This function essentially takes the union of all the provider flags
         // for a given provider GUID. This comes about when multiple providers
         // for the same GUID are provided and request different provider flags.
         // TODO: Only forward the calls that are requested to each provider.
         for (auto &provider : trace.providers_) {
-            if (providerFlags.find(provider.get().guid_) != providerFlags.end()) {
-                providerFlags[provider.get().guid_] = std::make_tuple<UCHAR, ULONGLONG, ULONGLONG, UCHAR> (0, 0, 0, 0);
+            if (provider_flags.find(provider.get().guid_) != provider_flags.end()) {
+                provider_flags[provider.get().guid_].flags_tuple_ = std::make_tuple<UCHAR, ULONGLONG, ULONGLONG, UCHAR> (0, 0, 0, 0);
             }
 
-            std::get<0>(providerFlags[provider.get().guid_]) |= provider.get().level_;
-            std::get<1>(providerFlags[provider.get().guid_]) |= provider.get().any_;
-            std::get<2>(providerFlags[provider.get().guid_]) |= provider.get().all_;
-            std::get<3>(providerFlags[provider.get().guid_]) |= provider.get().trace_flags_;
+            std::get<0>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().level_;
+            std::get<1>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().any_;
+            std::get<2>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().all_;
+            std::get<3>(provider_flags[provider.get().guid_].flags_tuple_) |= provider.get().trace_flags_;
+
+            for(const auto& filter : provider.get().filters_)
+            {
+                if (filter.provider_filter_event_id() > 0)
+                {
+                	//native id existing, set native filters
+                	provider_flags[provider.get().guid_].provider_filter_event_ids_.push_back(filter.provider_filter_event_id());
```

Comment:
this would benefit from being broken across two lines for better readability:


```
auto provider_filter_event_ids = provider_flags[provider.get().guid_].provider_filter_event_ids_;
provider_filter_event_ids = .push_back(filter.provider_filter_event_id()); 
```

Reply:
done